### ISSUE TITLE
svd test throwing exception on julia 1.6.1

### ISSUE THREAD

**Issue by bcgv** [JuliaArrays/StaticArrays.jl#925]
This test throws an exception on julia 1.6.1
```
svd: Error During Test at D:\xxx\.julia\dev\StaticArrays\test\svd.jl:80
  Test threw exception
  Expression: (#= D:\xxx\.julia\dev\StaticArrays\test\svd.jl:80 =# @inferred(svd_full_false(m_sing2))).S ≈ (svd(Matrix(m_sing2))).S
  return type StaticArrays.SVD{Float64, SMatrix{3, 2, Float64, 6}, SVector{2, Float64}, SMatrix{2, 2, Float64, 4}} does not match inferred return type Union{StaticArrays.SVD{Float64, SMatrix{3, 3, Float64, 9}, SVector{2, Float64}, SMatrix{2, 2, Float64, 4}}, StaticArrays.SVD{Float64, SMatrix{3, 2, Float64, 6}, SVector{2, Float64}, SMatrix{2, 2, Float64, 4}}}
  Stacktrace:
   [1] error(s::String)
     @ Base .\error.jl:33
   [2] macro expansion
     @ D:\xxx\.julia\dev\StaticArrays\test\svd.jl:80 [inlined]
   [3] macro expansion
     @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Test\src\Test.jl:1151 [inlined]
   [4] macro expansion
     @ D:\xxx\.julia\dev\StaticArrays\test\svd.jl:17 [inlined]
   [5] macro expansion
     @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Test\src\Test.jl:1151 [inlined]
   [6] top-level scope
     @ D:\xxx\.julia\dev\StaticArrays\test\svd.jl:4
```

**Comment by mateuszbaran** [comment]
I think the test is wrong, there should be `svd_full_false(A) = svd(A, full=false)` instead of `svd_full_false(A) = svd(m_sing2, full=false)`.

**Comment by thchr** [comment]
This was fixed by #928 and can be closed now.
